packages/components/src/components/dropdown/dropdown.tsx

```diff
@@ -325,7 +327,19 @@ const Dropdown = ({
         handleVisibility();
     };
 
-    const handleVisibility = () => {
+    const handleVisibility = (e?: React.MouseEvent<HTMLDivElement>) => {
+        if (should_open_on_hover) {
+            if (e?.type === 'mouseover') {
+                setIsListVisible(true);
+                return;
+            } else if (e?.type === 'mouseleave') {
+                setIsListVisible(false);
+                return;
+            }
+        } else if (e?.type === 'mouseover' || e?.type === 'mouseleave') {
+            return;
+        }
```

Comment:
```suggestion
        if (should_open_on_hover) {
            if (e?.type === 'mouseover') {
                setIsListVisible(true);
                return;
            } else if (e?.type === 'mouseleave') {
                setIsListVisible(false);
                return;
            }
        } 
        
        return;
```

would this be the same?

Reply:
Hi, let me check

Reply:
No, because `return` happens before part for click